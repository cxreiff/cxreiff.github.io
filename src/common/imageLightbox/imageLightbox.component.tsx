import { useState } from "react";
import { Lightbox } from "react-modal-image";
import cn from "classnames";

import { Image } from "src/common/image/image.component";

import styles from "./imageLightbox.module.scss";

type ImageLightboxProps = {
  aspect: number;
  small: string;
  medium: string;
  large: string;
  alt?: string;
  className?: string;
};

export const ImageLightbox = ({
  aspect,
  small,
  medium,
  large,
  alt,
  className,
}: ImageLightboxProps) => {
  const [opened, setOpened] = useState(false);

  return (
    <div className={cn(styles.image_lightbox, className)}>
      <div onClick={() => setOpened(true)}>
        <Image
          className={styles.interactive}
          aspect={aspect}
          source={small}
          alt={alt}
        />
      </div>
      {opened && (
        <Lightbox
          medium={medium}
          large={large}
          alt={alt}
          hideDownload={true}
          hideClose={true}
          onClose={() => {
            setOpened(false);
          }}
        />
      )}
    </div>
  );
};
