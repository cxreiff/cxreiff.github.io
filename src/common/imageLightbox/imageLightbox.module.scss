@use "../../shared.scss" as *;

.image_lightbox {
  width: 100%;
}

.interactive {
  cursor: pointer;
}
