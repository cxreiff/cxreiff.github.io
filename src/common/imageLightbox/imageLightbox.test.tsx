import { mount, ReactWrapper } from 'enzyme'

import { ImageLightbox } from './imageLightbox.component'

describe('imageLoader', () => {
    
    let component: ReactWrapper<typeof ImageLightbox, void>
    
    beforeEach(() => {
        component = mount<typeof ImageLightbox, void>(
            <ImageLightbox
                aspect={1/1}
                small={'test'}
                medium={'test'}
                large={'test'}
                alt={'test'}
                className={'test'}
            />
        )
    })
    
    it('should render', () => {
        expect(component).toMatchSnapshot()
    })

    it('should render lightbox', () => {
        component.find('div.test').simulate('click')
        expect(component).toMatchSnapshot()
    })

    it('should close lightbox', () => {
        component.find('div.test').simulate('click')
        component.find('.__react_modal_image__icon_menu').childAt(1).simulate('click')
        expect(component).toMatchSnapshot()
    })
})
