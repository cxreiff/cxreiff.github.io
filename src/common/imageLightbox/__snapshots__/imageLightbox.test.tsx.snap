// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should open lightbox 1`] = `
<div>
  <div
    class="image_lightbox"
  >
    <div>
      <div
        class="image interactive hidden"
      >
        <img
          alt="test image"
          src="test"
        />
        <svg
          viewBox="0 0 1.3333333333333333 1"
        />
      </div>
    </div>
    <div>
      <div
        class="__react_modal_image__modal_container"
      >
        <div
          class="__react_modal_image__modal_content"
        >
          <div>
            <svg
              fill="#ffffff"
              height="48"
              viewBox="0 0 24 24"
              width="48"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 2v6h.01L6 8.01 10 12l-4 4 .01.01H6V22h12v-5.99h-.01L18 16l-4-4 4-3.99-.01-.01H18V2H6zm10 14.5V20H8v-3.5l4-4 4 4zm-4-5l-4-4V4h8v3.5l-4 4z"
              />
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
            </svg>
            <img
              class="__react_modal_image__medium_img"
              id="react-modal-image-img"
              src="test"
              style="transform: translate3d(-50%, -50%, 0) rotate(0deg);"
            />
          </div>
        </div>
        <div
          class="__react_modal_image__header"
        >
          <span
            class="__react_modal_image__icon_menu"
          >
            <a>
              <svg
                fill="#ffffff"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M7 14H5v5h5v-2H7v-3zm-2-4h2V7h3V5H5v5zm12 7h-3v2h5v-5h-2v3zM14 5v2h3v3h2V5h-5z"
                />
              </svg>
            </a>
            <a>
              <svg
                fill="#ffffff"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </a>
          </span>
          <span
            class="__react_modal_image__caption"
          >
            test image
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render 1`] = `
<div>
  <div
    class="image_lightbox"
  >
    <div>
      <div
        class="image interactive hidden"
      >
        <img
          alt="test image"
          src="test"
        />
        <svg
          viewBox="0 0 1.3333333333333333 1"
        />
      </div>
    </div>
  </div>
</div>
`;
